### PR TITLE
Add command line lint manipulation in rustdoc

### LINKED ISSUE #50082: rustdoc: provide option to turn intra-link resolution errors into proper errors
When rendering docs for the standard library just now, i noticed some warnings come out of rustdoc. It would be prudent to make these force rustdoc to fail, so they would cause CI to fail. We did something similar during the Markdown transition with a temporary flag, `--deny-rendering-differences`, so i imagine it would work out the same.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #50082.

r? @QuietMisdreavus

### CODE REVIEW

**Comment by stokhos** [comment]
ping from triage @nikomatsakis @spastorino @Manishearth will you have time to review this PR in the near future?

File: src/librustdoc/core.rs [hunk 1]
```diff
@@ -213,7 +217,7 @@ pub fn run_core(search_paths: SearchPaths,
         } else {
             vec![]
         },
-        lint_cap: Some(lint::Forbid),
+        lint_cap: Some(lint_cap.unwrap_or_else(|| lint::Forbid)),
```
Comment (QuietMisdreavus) [inline src/librustdoc/core.rs]: You can leave this as `None` if you're going to default it to `Forbid` anyway; rustc will do this unwrap itself later on.

File: src/librustdoc/lib.rs [hunk 2]
```diff
@@ -304,6 +305,28 @@ pub fn opts() -> Vec<RustcOptGroup> {
...
+        unstable("forbid", |o| {
+            o.optmulti("F", "forbid", "Set lint forbidden", "OPT")
+        }),
+        unstable("cap-lints", |o| {
+            o.optmulti(
+                "",
+                "cap-lints",
+                "Set the most restrictive lint level. \
+                 More restrictive lints are capped at this \
+                 level. Byt default, it is at `forbid` level.",
```
Comment (QuietMisdreavus) [inline src/librustdoc/lib.rs]: Typo, "*byt* default"

**Review by QuietMisdreavus — COMMENTED** [review]
Looks good, thanks so much! Just a couple style things.

File: src/librustdoc/core.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/core.rs]: No, I did at first but didn't get the expected result.

Comment (QuietMisdreavus) [reply src/librustdoc/core.rs]: Huh, weird, I didn't realize it was unwrapped anywhere other than where it was used.

**Comment by QuietMisdreavus** [comment]
r=me pending travis

**Pull request merged**